src/FSharpVSPowerTools.Logic/LintTagger.fs

```diff
@@ -0,0 +1,95 @@
+﻿namespace FSharpVSPowerTools.Linting
+
+open System
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Text.Tagging
+open Microsoft.VisualStudio.Shell.Interop
+open FSharpVSPowerTools
+open FSharpVSPowerTools.AsyncMaybe
+open FSharpVSPowerTools.ProjectSystem
+open FSharpLint.Application
+
+type LintTag(tooltip) = 
+    inherit ErrorTag(Constants.LintTagErrorType, tooltip)
+
+type LintTagger(textDocument: ITextDocument,
+                view: ITextView, 
+                vsLanguageService: VSLanguageService, 
+                serviceProvider: IServiceProvider,
+                projectFactory: ProjectFactory) as self =
+    let tagsChanged = Event<_, _>()
+    let mutable wordSpans = []
+
+    let buffer = view.TextBuffer
+
+    let updateAtCaretPosition () =
+        asyncMaybe {
+            let dte = serviceProvider.GetService<EnvDTE.DTE, SDTE>()
+            let! doc = dte.GetCurrentDocument(textDocument.FilePath)
+            let! project = projectFactory.CreateForDocument buffer doc
+            let source = buffer.CurrentSnapshot.GetText()
+            let! checkResults = vsLanguageService.ParseAndCheckFileInProject (doc.FullName, source, project) |> liftAsync
+            let! ast = checkResults.GetUntypedAst()
+            let res = 
+                Lint.lintParsedFile 
+                    { FinishEarly = None
+                      Configuration = None //Configuration.defaultConfiguration
+                      ReceivedWarning = None }
+                    { Ast = ast
+                      Source = source
+                      TypeCheckResults = checkResults.GetCheckResults() }
+                    doc.FullName
+
+            return
+                match res with
+                | LintResult.Success warnings ->
+                    warnings 
+                    |> Seq.distinctBy (fun warn -> warn.Range, warn.Info)
+                    |> Seq.choose (fun warn -> 
+                        fromFSharpRange view.TextBuffer.CurrentSnapshot warn.Range
+                        |> Option.map (fun span -> warn, span))
+                    |> Seq.toList
+                | LintResult.Failure _ -> []
+        }
+        |> Async.map (fun spans -> 
+            let spans = 
+                match spans with 
+                | Some x -> x
+                | None -> []
+            wordSpans <- spans
+            let span = SnapshotSpan(buffer.CurrentSnapshot, 0, buffer.CurrentSnapshot.Length)
+            tagsChanged.Trigger(self, SnapshotSpanEventArgs span))
+        |> Async.StartInThreadPoolSafe
+
+    let docEventListener = new DocumentEventListener ([ViewChange.bufferEvent textDocument.TextBuffer], 200us, updateAtCaretPosition)
```

Comment:
Buffer event is not raised. I have no idea why. The code is identical to what's used in `SyntaxConstructClassifier`. Ideas?